src/editor/editor.cpp

```diff
@@ -537,14 +756,20 @@ Editor::set_level(std::unique_ptr<Level> level, bool reset)
   m_enabled = true;
 
   if (reset) {
-    m_toolbox_widget->get_tilebox().set_input_type(EditorTilebox::InputType::NONE);
+    m_toolbox_widget->get_tilebox().set_input_type(InputType::NONE);
   }
 
-  // Reload level.
-  m_level = nullptr;
   m_levelloaded = true;
 
-  m_level = std::move(level);
+  if (level != nullptr)
```

Comment:
Would this make more sense?:

```suggestion
  if (!m_temp_level)
```